packages/ide/src/client.ts

```diff
@@ -34,6 +34,11 @@ export abstract class IdeClient {
 		this.loadTime = time(2500);
 
 		let appWindow: Window | undefined;
+
+		window.onbeforeunload = () => {
```

Comment:
It might make sense to use `addEventListener` to stay consistent with the other event handlers.